src/pytest_workflow/workflow.py

```diff
@@ -17,54 +17,171 @@
 """
 Contains all functionality regarding the running of workflows and reporting
 on stdout, stderr and exit code.
-This file was created by A.H.B. Bollen
+This file was created by A.H.B. Bollen. Multithreading functionality was added
+later.
 """
-
+import queue
 import shlex
 import subprocess  # nosec: security implications have been considered
+import threading
+import time
 from pathlib import Path
+from typing import Optional  # noqa: F401  # used for typing
 
 
 class Workflow(object):
-
+    # pylint: disable=too-many-instance-attributes
```

Comment:
Meta-comment: Nearly all methods/properties do `self.wait()`. Perhaps using a decorator makes it it a bit clearer as to why this happens, and allows to completely refactor the code without touching the multithreading code.

Reply:
Added this to the to-do list #32